src/index.ts

```diff
@@ -779,6 +810,115 @@ export class NodeSSH {
     return !failed
   }
 
+  forwardIn(remoteAddr: string, remotePort: number, onConnection?: SSHForwardInListener): Promise<SSHForwardInDetails> {
+    const connection = this.getConnection()
+
+    return new Promise((resolve, reject) => {
+      connection.forwardIn(remoteAddr, remotePort, (error, port) => {
+        if (error) {
+          reject(error)
+          return
+        }
+
+        const handler: SSHForwardInListener = (details, acceptConnection, rejectConnection) => {
+          if (details.destIP === remoteAddr && details.destPort === port) {
+            onConnection?.(details, acceptConnection, rejectConnection)
+          }
+        }
+
+        if (onConnection) {
+          connection.on('tcp connection', handler)
+        }
+
+        const unforward = (): Promise<void> => {
+          return new Promise((_resolve, _reject) => {
+            connection.off('tcp connection', handler)
+            connection.unforwardIn(remoteAddr, port, (_error) => {
+              if (_error) {
+                _reject(error)
+              }
+
+              _resolve()
+            })
+          })
+        }
+
+        resolve({ port, unforward })
+      })
+    })
+  }
+
+  forwardOut(srcIP: string, srcPort: number, dstIP: string, dstPort: number): Promise<Channel> {
+    const connection = this.getConnection()
+
+    return new Promise((resolve, reject) => {
+      connection.forwardOut(srcIP, srcPort, dstIP, dstPort, (error, channel) => {
+        if (error) {
+          reject(error)
+          return
+        }
+
+        resolve(channel)
+      })
+    })
+  }
+
+  forwardInStreamLocal(
+    socketPath: string,
+    onConnection?: SSHForwardInStreamLocalListener,
+  ): Promise<SSHForwardInStreamLocalDetails> {
+    const connection = this.getConnection()
+
+    return new Promise((resolve, reject) => {
+      connection.openssh_forwardInStreamLocal(socketPath, (error) => {
+        if (error) {
+          reject(error)
+          return
+        }
+
+        const handler: SSHForwardInStreamLocalListener = (details, acceptConnection, rejectConnection) => {
+          if (details.socketPath === socketPath) {
+            onConnection?.(details, acceptConnection, rejectConnection)
+          }
+        }
+
+        if (onConnection) {
+          connection.on('unix connection', handler)
+        }
+
+        const unforward = (): Promise<void> => {
```

Comment:
Let's stick to the `Disposable` interface by calling this `dispose` instead. What do you think?